### PR TITLE
[drape] Moved Glyph out of GlyphManager

### PR DESCRIPTION
**Pull request by biodranik**
Also did small refactoring in a GlyphManager.

Based on #7855 that should be merged first.

### CODE REVIEW

File: drape/glyph_manager.cpp [hunk 1]
```diff
@@ -257,22 +215,22 @@ class Font
     while (gindex)
       charcodes.push_back(FT_Get_Next_Char(m_fontFace, charcodes.back(), &gindex));
 
-    sort(charcodes.begin(), charcodes.end());
-    charcodes.erase(unique(charcodes.begin(), charcodes.end()), charcodes.end());
+    std::sort(charcodes.begin(), charcodes.end());
+    charcodes.erase(std::unique(charcodes.begin(), charcodes.end()), charcodes.end());
```
Comment (vng) [inline drape/glyph_manager.cpp:219]: nit: base::SortUnique();

File: drape/glyph_manager.cpp [hunk 2]
```diff
@@ -346,6 +304,19 @@ using TUniBlockIter = TUniBlocks::const_iterator;
 
 struct GlyphManager::Impl
 {
+  Impl() = default;
+
+  ~Impl() {
```
Comment (vng) [inline drape/glyph_manager.cpp:309]: { from new line

File: drape/glyph_manager.cpp [hunk 3]
```diff
@@ -125,57 +113,39 @@ class Font
...
       MYTHROW(InvalidFontException, (g_FT_Errors[err].m_code, g_FT_Errors[err].m_message));
   }
 
-  bool IsValid() const
+  ~Font()
   {
-    return m_fontFace != nullptr && m_fontFace->num_glyphs > 0;
+    ASSERT(m_fontFace, ());
+    FREETYPE_CHECK(FT_Done_Face(m_fontFace));
+    m_fontFace = nullptr;
```
Comment (vng) [inline drape/glyph_manager.cpp:141]: ```m_fontFace = nullptr;```  not needed in dtor

Comment (vng) [reply drape/glyph_manager.cpp:141]: If so, we should make ```DISALLOW_COPY_AND_MOVE(Font)```

**Review by vng — CHANGES_REQUESTED** [review]

**Comment by biodranik** [comment]
@vng fixed, PTAL

**Review by vng — APPROVED** [review]

**Pull request merged**